src/realm/table.hpp

```diff
@@ -1442,6 +1461,125 @@ class Table {
     friend class Group;
 };
 
+/*****************************************************************************/
+
+template <class T>
+using ListRef = std::unique_ptr<Table::List<T>>;
+template <class T>
+using ConstListRef = std::unique_ptr<const Table::List<T>>;
+
+template <typename T>
+class Table::List {
+public:
+    class Iterator : public std::iterator<std::input_iterator_tag, T> {
```

Comment:
`std::iterator` is deprecated. This is currently a ForwardIterator and not just an InputIterator, and it could be a RandomAccessIterator.